Authentication/ExportAuthenticator.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace EzSystems\RecommendationBundle\Authentication;
+
+use eZ\Publish\Core\REST\Common\Exceptions\NotFoundException;
+use EzSystems\RecommendationBundle\Helper\FileSystem;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class ExportAuthenticator implements Authenticator
+{
+    /** @var \Symfony\Component\HttpFoundation\RequestStack */
+    private $requestStack;
+
+    /** @var \EzSystems\RecommendationBundle\Helper\FileSystem */
+    private $fileSystem;
+
+    /** @var bool */
+    private $method;
+
+    /** @var string */
+    private $login;
+
+    /** @var string */
+    private $password;
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+     * @param \EzSystems\RecommendationBundle\Helper\FileSystem $fileSystem
+     * @param string $method
+     * @param string $login
+     * @param string $password
+     */
+    public function __construct(
+        RequestStack $requestStack,
+        FileSystem $fileSystem,
+        $method,
+        $login,
+        $password
+    ) {
+        $this->requestStack = $requestStack;
+        $this->fileSystem = $fileSystem;
+        $this->method = $method;
+        $this->login = $login;
+        $this->password = $password;
+    }
+
+    /**
+     * Returns current authentication method and credentials.
+     *
+     * @return array
+     */
+    public function getCredentials()
+    {
+        return array(
+            'method' => $this->method,
+            'login' => $this->login,
+            'password' => $this->password,
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function authenticate()
+    {
+        $server = $this->requestStack->getCurrentRequest()->server;
+
+        if ($this->method == 'none') {
+            return true;
+        }
+
+        if (!empty($this->login)
+            && !empty($this->password)
+            && $server->get('PHP_AUTH_USER') === $this->login
+            && $server->get('PHP_AUTH_PW') === $this->password
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param $filePath
+     *
+     * @return bool
+     */
+    public function authenticateByFile($filePath)
```

Comment:
I was not aware of the authentication method. Is there a documentation page?

Reply:
This feature was introduced in original PR. When authentication method is set to `basic`, the export will generate the file `.htpasswd`, every export has unique password generated. You can send headers with user and password by basicauth and passed values will be compared to values in `.htpasswd` file. The authentication mechanism is resolved by php code, so even if it looks like apache only solution - it will work with all types of web server, not only apache